src/operation_cost.rs

```diff
@@ -0,0 +1,142 @@
+use std::collections::HashMap;
+
+use anyhow::{anyhow, Result};
+use apollo_compiler::{values::Selection, ApolloCompiler};
+
+use crate::compiler_ext::CompilerAdditions;
+
+struct Context<'a> {
+    compiler: &'a ApolloCompiler,
+    cost_map: &'a HashMap<String, i32>,
+}
+
+pub fn operation_cost(
+    sdl: &String,
+    operation: &String,
+    operation_name: Option<&String>,
+    cost_map: &HashMap<String, i32>,
+) -> Result<i32> {
+    let mut input = sdl.to_owned();
+    input.push_str(operation.as_str());
+
+    let compiler = ApolloCompiler::new(&input);
```

Comment:
You almost never want `&String` as a type. If you run `cargo clippy` it tells you why. I think it's slightly neater a single string from format (thus immutable) than to concatenate, but it's not a big thing.

```suggestion
pub fn operation_cost(
    sdl: &str,
    operation: &str,
    operation_name: Option<&String>,
    cost_map: &HashMap<String, i32>,
) -> Result<i32> {
    let input = format!("{}\n{}", sdl, operation);

    let compiler = ApolloCompiler::new(&input);
```